File: psychromol/units.py
from __future__ import annotations

PRESSURE_UNITS: dict[str, float] = {"kPa": 1.0, "hPa": 0.1, "Pa": 0.001}


def kpa_from(value: float, unit: str) -> float:
    try:
        factor = PRESSURE_UNITS[unit]
    except KeyError as exc:
        raise ValueError(f"unknown pressure unit {unit!r}") from exc
    return value * factor

File: frontend/js/units.js
export const PA_PER_KPA = 1000;

export const PRESSURE_UNITS = Object.freeze({ kPa: 1, hPa: 0.1, Pa: 0.001 });

export function paFromKpa(kpa) {
  return kpa * PA_PER_KPA;
}

export function kpaFromPa(pa) {
  return pa / PA_PER_KPA;
}

export function kpaFrom(value, unit) {
  const factor = PRESSURE_UNITS[unit];
  if (factor === undefined) throw new Error(`unknown pressure unit ${unit}`);
  return value * factor;
}
